lib/ansible/modules/cloud/amazon/lambda_policy.py

```diff
@@ -0,0 +1,494 @@
+    current_state = 'absent'
+    state = module.params['state']
+    action_taken = 'none'
+
+    # check if the policy exists
+    current_policy_statement = get_policy_statement(module, aws)
+    if current_policy_statement:
+        current_state = 'present'
+
+    if state == 'present':
+        if current_state == 'present':
+            # check if policy has changed and update if necessary
+            # since there's no API to update a policy statement, it must first be removed
+            if not policy_equal(module, current_policy_statement):
+                remove_policy_permission(module, aws)
+                changed = add_policy_permission(module, aws)
+                action_taken = 'updated'
+        else:
+            # add policy statement
+            changed = add_policy_permission(module, aws)
+            action_taken = 'added'
+    else:
+        if current_state == 'present':
+            # remove the policy statement
+            changed = remove_policy_permission(module, aws)
+            action_taken = 'deleted'
+
+    return dict(changed=changed, ansible_facts=dict(lambda_policy_action=action_taken))
+
+
+# ---------------------------------------------------------------------------------------------------
+#
+#   MAIN
+#
+# ---------------------------------------------------------------------------------------------------
+
+def main():
+    """
+    Main entry point.
+
+    :return dict: ansible facts
+    """
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(required=False, default='present', choices=['present', 'absent']),
+            function_name=dict(required=True, default=None, aliases=['lambda_function_arn', 'function_arn']),
+            statement_id=dict(required=True, default=None, aliases=['sid']),
+            alias=dict(required=False, default=None),
+            version=dict(type='int', required=False, default=0),
+            action=dict(required=True, default=None),
+            principal=dict(required=True, default=None),
+            source_arn=dict(required=False, default=None),
+            source_account=dict(required=False, default=None),
+            event_source_token=dict(required=False, default=None),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[['alias', 'version'],
+                            ['event_source_token', 'source_arn'],
+                            ['event_source_token', 'source_account']],
+        required_together=[]
+    )
+
+    # validate dependencies
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required for this module.')
+
+    aws = AWSConnection(module, ['lambda'])
+
+    validate_params(module, aws)
+
+    results = manage_state(module, aws)
+
+    module.exit_json(**results)
+
+
+# ansible import module(s) kept at ~eof as recommended
+from ansible.module_utils.basic import *
```

Comment:
Should be changed to new style imports. Also then no need to set HAS_BOTO3, you can import it.